openhgnn/models/SeHGNN.py

```diff
@@ -0,0 +1,323 @@
+            elif self.cformat == 'channel-last':
+                return torch.einsum('bmc,mn->bnc', x, self.W) + self.bias.T
+            else:
+                assert False
+        else:
+            if self.cformat == 'channel-first':
+                return torch.einsum('bcm,cmn->bcn', x, self.W) + self.bias
+            elif self.cformat == 'channel-last':
+                return torch.einsum('bmc,cmn->bnc', x, self.W) + self.bias.T
+            else:
+                assert False
+
+
+class L2Norm(nn.Module):
+
+    def __init__(self, dim):
+        super(L2Norm, self).__init__()
+        self.dim = dim
+
+    def forward(self, x):
+        return F.normalize(x, p=2, dim=self.dim)
+
+@register_model('SeHGNN')
+class SeHGNN(BaseModel):
+    r"""
+    This is a model SimpleHGN from `Simple and Efficient Heterogeneous Graph Neural Network
+    <https://doi.org/10.48550/arXiv.2207.02547>`__
+    This model is a metapath-based model. It put the neighbor aggregation in the preprocessing step, and using
+    the single-layer structure and long metapaths. It performed over the state-of-the-arts on both accuracy and training speed.
+
+    the neighbor aggregation
+    .. math::
+        \mathrm{X}^{P} = \hat{A}_{c,c_{1}}\hat{A}_{c_{1},c_{2}}...\hat{A}_{c_{l-1},c_{l}} \mathrm{X}^{c_{l}}
+    feature projection
+    .. math::
+        {\mathrm{H}^{'}}^{P} = MLP_{P}(\mathrm{X}^{P})
+    semantic fusion (transformer):
+    .. math::
+        q^{\mathcal{P}_{i}}=W_{Q} h^{\prime \mathcal{P}_{i}}, k^{\mathcal{P}_{i}}=W_{K} h^{\prime \mathcal{P}_{i}}, v^{\mathcal{P}_{i}}=W_{V} h^{\prime \mathcal{P}_{i}}, \mathcal{P}_{i} \in \Phi_{X} \\
+    .. math::
+        \alpha_{\left(\mathcal{P}_{i}, \mathcal{P}_{j}\right)}=\frac{\exp \left(q^{\mathcal{P}_{i}} \cdot k^{{\mathcal{P}_{j}}^{T}}\right)}{\sum_{\mathcal{P}_{t} \in \Phi_{X}} \exp \left(q^{\mathcal{P}_{i}} \cdot k^{{\mathcal{P}_{t}}^{T}}\right)}
+    .. math::
+        h^{\mathcal{P}_{i}}=\beta \sum_{\mathcal{P}_{j} \in \Phi_{X}} \alpha_{\left(\mathcal{P}_{i}, \mathcal{P}_{j}\right)} v^{\mathcal{P}_{j}}+h^{\prime \mathcal{P}_{i}}
+    """
+    @classmethod
+    def build_model_from_args(cls, args):
+        return cls(args)
+
+    def __init__(self, args):
+        super(SeHGNN, self).__init__()
+        dataset = args.dataset
+        data_size = args.data_size
+        nfeat = args.nfeat
+        hidden = args.hidden
+        nclass = args.nclass
+        num_feats = args.num_feats
+        num_label_feats = args.num_label_feats
+        tgt_key = args.tgt_key
+        dropout = args.dropout
+        input_drop = args.input_drop
+        att_drop = args.att_drop
+        label_drop = args.label_drop
+        n_layers_1 = args.n_layers_1
+        n_layers_2 = args.n_layers_2
+        n_layers_3 = args.n_layers_3
+        act = args.act
+        residual = args.residual
+        bns = args.bns
+        label_bns = args.label_bns
+        label_residual = args.label_residual
```

Comment:
Do not write the interface like this. If someone wants to use SeHGNN with other arguments, he will get an `Error` report! Please refer to the interface of the other models.